forge-game/src/main/java/forge/game/ability/effects/AnimateEffect.java

```diff
@@ -295,24 +295,27 @@ protected String getStackDescription(SpellAbility sa) {
         }
         // sb.append(abilities)
         // sb.append(triggers)
-        if (!permanent) {
+        if (!permanent && sb.length() > initial) {
             final String duration = sa.getParam("Duration");
             if ("UntilEndOfCombat".equals(duration)) {
-                sb.append("until end of combat.");
+                sb.append("until end of combat");
             } else if ("UntilHostLeavesPlay".equals(duration)) {
-                sb.append("until ").append(host).append(" leaves the battlefield.");
+                sb.append("until ").append(host).append(" leaves the battlefield");
             } else if ("UntilYourNextUpkeep".equals(duration)) {
-                sb.append("until your next upkeep.");
+                sb.append("until your next upkeep");
             } else if ("UntilYourNextTurn".equals(duration)) {
-                sb.append("until your next turn.");
+                sb.append("until your next turn");
             } else if ("UntilControllerNextUntap".equals(duration)) {
-                sb.append("until its controller's next untap step.");
+                sb.append("until its controller's next untap step");
             } else {
-                sb.append("until end of turn.");
+                sb.append("until end of turn");
             }
-        } else {
-            sb.append(".");
         }
+        if (sa.getParam("staticAbilities").contains("MustAttack")) {
```

Comment:
to fix NPE, check hasparam.. `if (sa.hasParam("staticAbilities"))`

Reply:
I think this should do it - thanks!